src/cryptoadvance/specter/server_endpoints/wallets_api.py

```diff
@@ -0,0 +1,933 @@
+            mimetype="text/csv",
+        )
+        # add a filename
+        response.headers.set(
+            "Content-Disposition", "attachment", filename="addresses_list.csv"
+        )
+        return response
+    except Exception as e:
+        handle_exception(e)
+        flash(_("Failed to export addresses list. Error: {}").format(e), "error")
+        return redirect(url_for("index"))
+
+
+# Export wallet transaction history
+@wallets_endpoint_api.route("/wallet/<wallet_alias>/transactions.csv")
+@login_required
+def tx_history_csv(wallet_alias):
+    wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
+    validate_merkle_proofs = app.specter.config.get("validate_merkle_proofs", False)
+    txlist = wallet.txlist(validate_merkle_proofs=validate_merkle_proofs)
+    search = request.args.get("search", None)
+    sortby = request.args.get("sortby", "time")
+    sortdir = request.args.get("sortdir", "desc")
+    txlist = json.loads(
+        process_txlist(
+            txlist, idx=0, limit=0, search=search, sortby=sortby, sortdir=sortdir
+        )["txlist"]
+    )
+    includePricesHistory = request.args.get("exportPrices", "false") == "true"
+
+    # stream the response as the data is generated
+    response = Response(
+        txlist_to_csv(wallet, txlist, app.specter, current_user, includePricesHistory),
+        mimetype="text/csv",
+    )
+    # add a filename
+    response.headers.set(
+        "Content-Disposition", "attachment", filename="transactions.csv"
+    )
+    return response
+
+
+# Export wallet UTXO list
+@wallets_endpoint_api.route("/wallet/<wallet_alias>/utxo.csv")
+@login_required
+def utxo_csv(wallet_alias):
+    try:
+        wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
+        includePricesHistory = request.args.get("exportPrices", "false") == "true"
+        search = request.args.get("search", None)
+        sortby = request.args.get("sortby", "time")
+        sortdir = request.args.get("sortdir", "desc")
+        txlist = json.loads(
+            process_txlist(
+                wallet.full_utxo,
+                idx=0,
+                limit=0,
+                search=search,
+                sortby=sortby,
+                sortdir=sortdir,
+            )["txlist"]
+        )
+        # stream the response as the data is generated
+        response = Response(
+            txlist_to_csv(
+                wallet,
+                txlist,
+                app.specter,
+                current_user,
+                includePricesHistory,
+            ),
+            mimetype="text/csv",
+        )
+        # add a filename
+        response.headers.set("Content-Disposition", "attachment", filename="utxo.csv")
+        return response
+    except Exception as e:
+        handle_exception(e)
+        return _("Failed to export wallet utxo. Error: {}").format(e), 500
+
+
+@wallets_endpoint_api.route("/wallet/<wallet_alias>/asset_balances")
```

Comment:
Looks like this needs `methods=["POST"]` specified to work with the JS.

Seeing:
```
[2021-12-13 09:22:36,500] INFO in _internal: 127.0.0.1 - - [13/Dec/2021 09:22:36] "POST /wallets/wallet/seedsigner_demo/asset_balances HTTP/1.1" 302 -
[2021-12-13 09:22:36,508] INFO in _internal: 127.0.0.1 - - [13/Dec/2021 09:22:36] "GET /wallets/wallet/seedsigner_demo/asset_balances HTTP/1.1" 200 -
```

Which seems to trigger the "Session expired. Please refresh and try again" error on the _next_ url.